internal/display/bufwindow.go

```diff
@@ -514,28 +539,45 @@ func (w *BufWindow) displayBuffer() {
 				}
 			}
 
-			if r != '\t' && r != ' '{
+			if r != '\t' && r != ' ' {
 				return r, style, bgoverridable
 			}
 
-			var returnrune rune
-			if r == '\t' {
-				indentrunes := []rune(b.Settings["indentchar"].(string))
-				// if empty indentchar settings, use space
-				if len(indentrunes) == 0 {
-					indentrunes = []rune{' '}
+			var indentrunes []rune
+			switch r {
+			case '\t':
+				if bloc.X < leadingwsEnd {
+					if indenttabchars != "" {
+						indentrunes = []rune(indenttabchars)
+					} else {
+						indentrunes = []rune(tabchars)
```

Comment:
Instead of having this `else` branch which duplicates the outer `else` branch below, we can just add `indenttabchars != ""` to the outer `if` condition?

Ditto for `spacechars` below.